engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -750,6 +752,11 @@ private void execute() throws YarnException, IOException
         expiryTime = StramUserLogin.refreshTokens(tokenLifeTime, FileUtils.getTempDirectoryPath(), applicationId, conf, hdfsKeyTabFile, credentials, rmAddress, true);
       }
 
+      if (System.currentTimeMillis() > nodeReportUpdateTime) {
```

Comment:
I have put it inside loop so that node reports get updated periodically. This helps in keeping latest available resources in node reports so that on container kill/restart, we can allocate nodes based on latest available resources.